tests/foreman/ui_airgun/test_remoteexecution.py

```diff
@@ -0,0 +1,275 @@
+
+
+@tier3
+def test_positive_run_scheduled_job_template_by_ip(session, module_vm_client_by_ip):
+    """Schedule a job to be ran against a host by ip
+
+    :id: 4387bed9-969d-45fb-80c2-b0905bb7f1bd
+
+    :Setup: Use pre-defined job template.
+
+    :Steps:
+
+        1. Set remote_execution_connect_by_ip on host to true
+        2. Navigate to an individual host and click Run Job
+        3. Select the job and appropriate template
+        4. Select "Schedule future execution"
+        5. Enter a desired time for the job to run
+        6. Click submit
+
+    :expectedresults:
+
+        1. Verify the job was not immediately ran
+        2. Verify the job was successfully ran after the designated time
+
+    :CaseLevel: System
+    """
+    job_time = 5 * 60
+    hostname = module_vm_client_by_ip.hostname
+    with session:
+        plan_time = session.browser.get_client_datetime() + datetime.timedelta(seconds=job_time)
+        assert session.host.search(hostname)[0]['Name'] == hostname
+        job_status = session.host.schedule_remote_job(
+            [hostname],
+            {
+                'job_category': 'Commands',
+                'job_template': 'Run Command - SSH Default',
+                'template_content.command': 'ls',
+                'schedule': 'Schedule future execution',
+                'schedule_content.start_at': plan_time.strftime("%Y-%m-%d %H:%M"),
+            },
+            wait_for_results=False,
+        )
+        # Note that to create this host scheduled job we spent some time from that plan_time, as it
+        # was calculated before creating the job
+        job_left_time = (plan_time - datetime.datetime.now()).total_seconds()
+        # assert that we have time left to wait, otherwise we have to use more job time,
+        # the job_time must be significantly greater than job creation time.
+        assert job_left_time > 0
+        assert job_status['overview']['hosts_table'][0]['Host'] == hostname
+        assert job_status['overview']['hosts_table'][0]['Status'] == 'N/A'
+        # sleep 3/4 of the left time
+        time.sleep(job_left_time * 3/4)
+        job_status = session.jobinvocation.read('Run ls', hostname, 'overview.hosts_table')
+        assert job_status['overview']['hosts_table'][0]['Host'] == hostname
+        assert job_status['overview']['hosts_table'][0]['Status'] == 'N/A'
+        # recalculate the job left time to be more accurate
+        job_left_time = (plan_time - datetime.datetime.now()).total_seconds()
+        # the last read time should not take more than 1/4 of the last left time
+        assert job_left_time > 0
+        # sleep the rest the job left time and ensure that the host job status is "success"
+        # but as we will be at the exact launch time and to let enough time to the UI to be
+        # notified, wait for additional 20 seconds (note that the UI event scheduler has a 10
+        # second period)
+        time.sleep(job_left_time + 20)
```

Comment:
I am really don't like that construct. We waited 3/4 of time left after creation is done, so what is a problem to wait 1/4 more?

Reply:
if we wait for 1/4 we would ignore the fact that we have read and spent time , but if we wait for new recalcuted left time to plan_time, after sleep we should be at the exact plan_time and if we add additional time we are in control about how much time we are adding in plus of plan_time.

Reply:
pretty much fine with a logic, but not fine with static number. Any alternative?